packages/jest-runtime/src/index.ts

```diff
@@ -560,10 +570,49 @@ class Runtime {
     }
   }
 
+  async collectV8Coverage() {
+    this._v8CoverageInstrumenter = new CoverageInstrumenter();
+
+    await this._v8CoverageInstrumenter.startInstrumenting();
+  }
+
+  async stopCollectingV8Coverage() {
+    if (!this._v8CoverageInstrumenter) {
+      throw new Error('You need to call `collectV8Coverage` first.');
+    }
+    this._v8CoverageResult = await this._v8CoverageInstrumenter.stopInstrumenting();
+  }
+
   getAllCoverageInfoCopy() {
     return deepCyclicCopy(this._environment.global.__coverage__);
   }
 
+  getAllV8CoverageInfoCopy(): V8CoverageResult {
+    if (!this._v8CoverageResult) {
+      throw new Error('You need to `stopCollectingV8Coverage` first');
+    }
+
+    return (
+      this._v8CoverageResult
+        .filter(res => res.url.startsWith('file://'))
+        .map(res => ({...res, url: fileURLToPath(res.url)}))
+        // TODO: will this work on windows? It might be better if `shouldInstrument` deals with it anyways
+        .filter(res => res.url.startsWith(this._config.rootDir))
+        .filter(res => this._fileTransforms.has(res.url))
+        .filter(res =>
```

Comment:
Can we at least fold these 3 filters into a single one? Generating a copy is potentially expensive, so maybe worth to optimize no of operations a bit

Reply:
Can do! I also need to filter out "internal" modules

Reply:
done